python/packages/autogen-ext/src/autogen_ext/tools/azure/_ai_search.py

```diff
@@ -0,0 +1,1616 @@
+        )
+
+    @overload
+    @classmethod
+    def load_component(
+        cls, model: Union[ComponentModel, Dict[str, Any]], expected: None = None
+    ) -> "BaseAzureAISearchTool": ...
+
+    @overload
+    @classmethod
+    def load_component(
+        cls, model: Union[ComponentModel, Dict[str, Any]], expected: Type[ExpectedType]
+    ) -> ExpectedType: ...
+
+    @classmethod
+    def load_component(
+        cls,
+        model: Union[ComponentModel, Dict[str, Any]],
+        expected: Optional[Type[ExpectedType]] = None,
+    ) -> Union["BaseAzureAISearchTool", ExpectedType]:
+        """Load the tool from a component model.
+
+        Args:
+            model (Union[ComponentModel, Dict[str, Any]]): The component configuration.
+            expected (Optional[Type[ExpectedType]]): Optional component class for deserialization.
+
+        Returns:
+            Union[BaseAzureAISearchTool, ExpectedType]: An instance of the tool.
+
+        Raises:
+            ValueError: If the component configuration is invalid.
+        """
+        if expected is not None and not issubclass(expected, BaseAzureAISearchTool):
+            raise TypeError(f"Cannot create instance of {expected} from AzureAISearchConfig")
+
+        target_class = expected if expected is not None else cls
+        assert hasattr(target_class, "_from_config"), f"{target_class} has no _from_config method"
+
+        if isinstance(model, ComponentModel) and hasattr(model, "config"):
+            config_dict = model.config
+        elif isinstance(model, dict):
+            config_dict = model
+        else:
+            raise ValueError(f"Invalid component configuration: {model}")
+
+        config = AzureAISearchConfig(**config_dict)
+
+        tool = target_class._from_config(config)
+        if expected is None:
+            return tool
+        return cast(ExpectedType, tool)
+
+    @property
+    def schema(self) -> ToolSchema:
+        """Return the schema for the tool."""
+        return {
+            "name": self.name,
+            "description": self.description,
+            "parameters": {
+                "type": "object",
+                "properties": {"query": {"type": "string", "description": "Search query text"}},
+                "required": ["query"],
+            },
+        }
+
+    def return_value_as_string(self, value: SearchResults) -> str:
+        """Convert the search results to a string representation.
+
+        This method is used to format the search results in a way that's suitable
+        for display to the user or for consumption by language models.
+
+        Args:
+            value (List[SearchResult]): The search results to convert.
+
+        Returns:
+            str: A formatted string representation of the search results.
+        """
+        if not value.results:
+            return "No results found."
+
+        result_strings: List[str] = []
+        for i, result in enumerate(value.results, 1):
```

Comment:
We can use a pretty-printed JSON here with indent=2. The JSON schema of the tool already indicates the structure of the returned value, and language models understand the structure of JSON. So we can just return a pretty-printed JSON string.

Reply:
Doesn't look like the new method _format_result_for_display is used?